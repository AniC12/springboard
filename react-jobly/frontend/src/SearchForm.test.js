import React from "react";
import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchForm from "./SearchForm";

it("matches snapshot", function () {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("updates input field on change", () => {
    const { getByPlaceholderText } = render(<SearchForm searchFor={() => { }} />);
    const input = getByPlaceholderText("Enter search term..");

    userEvent.type(input, "test");
    expect(input.value).toBe("test");
});

it("calls searchFor function on form submission", () => {
    const mockSearchFor = jest.fn();
    const { getByPlaceholderText, getByText } = render(<SearchForm searchFor={mockSearchFor} />);

    const input = getByPlaceholderText("Enter search term..");
    const submitButton = getByText("Submit");

    userEvent.type(input, "test");
    fireEvent.click(submitButton);

    expect(mockSearchFor).toHaveBeenCalledWith("test");
});

it("clears input after form submission", () => {
    const { getByPlaceholderText, getByText } = render(<SearchForm searchFor={() => { }} />);
    const input = getByPlaceholderText("Enter search term..");
    const submitButton = getByText("Submit");

    userEvent.type(input, "test");
    fireEvent.click(submitButton);

    expect(input.value).toBe("");
});


