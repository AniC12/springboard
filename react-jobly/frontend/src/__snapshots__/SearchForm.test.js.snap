// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <input
        name="searchTerm"
        placeholder="Enter search term.."
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
